src/app/modules/notification/services/notification.service.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { NotificationService } from './notification.service';
+
+describe('NotificationService', () => {
+  let service: NotificationService;
+  beforeEach(() => {
+    service = new NotificationService();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('push()', () => {
+    it('should add a notification to the list with an id', () => {
+      service.push('Hello world!');
+
+      expect(service.notifications.length).toEqual(1);
+
+      service.notifications.forEach((n) => {
+        expect(n.id).toBeDefined();
+      });
+    });
+
+    it('should merge notification options with default options', () => {
+      service.push('Hello world!', 'default', { isSelfClosing: false, isDismissable: false });
+
+      service.notifications.forEach((n) => {
+        expect(n.options).toEqual({
+          isSelfClosing: n.options.isSelfClosing,
+          isDismissable: n.options.isDismissable,
+          duration: service.options.duration
+        });
+      });
+    });
+  });
+
+  describe('dismiss()', () => {
+    it('should remove the notification', () => {
```

Comment:
it's maybe interesting to insert two notification and try to remove only one

Reply:
Better ?
```ts
describe('dismiss()', () => {
  it('should remove the notification', () => {
    service.push('First');
    service.push('Second');
    service.push('Third');

    const secondNotification = service.notifications[1];
    service.dismiss(secondNotification);

    setTimeout(() => {
      expect(service.notifications.length).toEqual(2);
      expect(service.notifications[0].message).toEqual('First');
      expect(service.notifications[1].message).toEqual('Third');
    }, service.removingDelay);
  });
});
```

Reply:
Yes ! It sounds good 😄

Reply:
Except the const, you can put directly `service.dismiss(service.notifications[1]);`
It's understandable that is the second notification

Reply:
Very nice Marie